dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ProcessingManager.java

```diff
@@ -93,6 +93,7 @@ private ProcessingManager(Configuration config) throws InvalidConfigurationExcep
                 RemoveDeadCondJumps.fromConfig(config),
                 reduceSymmetry ? SymmetryReduction.fromConfig(config) : null,
                 MemoryAllocation.newInstance(),
+                MemCpyUnrolling.newInstance(),
```

Comment:
We cannot do this so late. Not unrolling`MemCopy` during/after compilation will mess up at least spinloop detection.
At the same time, I think you want to do it after CP, so that you know how much to unroll. 
This also stands in contrast to what I said earlier about the data regs, i.e. that you want to throw an exception there.

I think fixing this issue requires a more general change in our infrastructure:
- `MemCpy` needs to be a `MemEvent`, likely tagged as both `Write` and `Load`
- `MemEvents` need to be have an array/list of addresses they access (`MemCpy` accesses 2 addresses)
- We may want to have a `getAddrRegs` method in addition to `getDataRegs`
- We need to revise all passes/analysis to make sure that they can deal with this.